src/gui/main.cpp

```diff
@@ -1,12 +1,66 @@
 #include "mainwindow.h"
+#include <QtCore/QtDebug>
+#include <QtCore/QtGlobal>
 #include <QtWidgets/QApplication>
 
+#include "log.hpp"
+
+std::shared_ptr<spdlog::logger> kDefaultLogger;
+std::shared_ptr<spdlog::logger> qtDefaultLogger;
+
+const std::string localFile(const std::string& fullPath)
+{
+    return std::strchr(fullPath.c_str(), '/') ? std::strrchr(fullPath.c_str(), '/') + 1 : fullPath;
+}
+
+void setupLogger(bool verbose)
+{
+    kDefaultLogger = spdlog::stdout_color_mt("cds");
+    qtDefaultLogger = spdlog::stdout_color_mt("cds-qt");
+    if (verbose) {
+        kDefaultLogger->set_level(spdlog::level::debug);
+        qtDefaultLogger->set_level(spdlog::level::debug);
+    }
+
+    qInstallMessageHandler([](QtMsgType type, const QMessageLogContext& context, const QString& msg) {
+        QByteArray localMsg = msg.toLocal8Bit();
+        switch (type) {
+        case QtDebugMsg:
+            qtDefaultLogger->debug(
+                "[{}@{}] {}", context.file ? localFile(context.file) : "", context.line, msg.toStdString());
+            break;
+        case QtInfoMsg:
+            qtDefaultLogger->info(
+                "[{}@{}] {}", context.file ? localFile(context.file) : "", context.line, msg.toStdString());
+            break;
+        case QtWarningMsg:
+            qtDefaultLogger->warn(
+                "[{}@{}] {}", context.file ? localFile(context.file) : "", context.line, msg.toStdString());
+            break;
+        case QtCriticalMsg:
+            qtDefaultLogger->critical(
+                "[{}@{}] {}", context.file ? localFile(context.file) : "", context.line, msg.toStdString());
+            break;
+        case QtFatalMsg:
+            qtDefaultLogger->error(
+                "[{}@{}] {}", context.file ? localFile(context.file) : "", context.line, msg.toStdString());
+            std::abort();
+        }
+    });
+}
+
 int main(int argc, char* argv[])
 {
     QApplication a(argc, argv);
-    //    QCoreApplication::setApplicationName("MDI Example");
-    //    QCoreApplication::setOrganizationName("QtProject");
-    //    QCoreApplication::setApplicationVersion(QT_VERSION_STR);
+#ifdef CDS_DEBUG
```

Comment:
I don't like this ifdef (as you probably know!) :) . What about doing something like that instead:

setupLogger(CDS_DEBUG)

This will require change of setupLogger definition to setupLogger(bool verbose=true);